src/gennylib/src/PoolManager.cpp

```diff
@@ -21,6 +21,7 @@ namespace {
 
 auto createPool(const std::string& name,
                 PoolManager::OnCommandStartCallback& apmCallback,
+                std::shared_ptr<EncryptionContext> encryption,
```

Comment:
It feels a bit odd that the other configuration is read from the context here, but the encryption is passed in?

I think that reading the EncryptionContext here would significantly simplify the code and lower the scale of the changes.